apps/server/test/lexical/server/provider/handlers/hover_test.exs

```diff
@@ -0,0 +1,117 @@
+defmodule Lexical.Server.Provider.Handlers.HoverTest do
+  alias Lexical.Document
+  alias Lexical.Proto.Convert
+  alias Lexical.Protocol.Requests
+  alias Lexical.Protocol.Types
+  alias Lexical.Server
+  alias Lexical.Server.Provider.Env
+  alias Lexical.Server.Provider.Handlers
+
+  import Lexical.Test.Protocol.Fixtures.LspProtocol
+  import Lexical.RemoteControl.Api.Messages
+  import Lexical.Test.Fixtures
+
+  use ExUnit.Case, async: false
+
+  setup_all do
+    project = project()
+
+    {:ok, _} = start_supervised(Document.Store)
+    {:ok, _} = start_supervised({DynamicSupervisor, Server.Project.Supervisor.options()})
+    {:ok, _} = start_supervised({Server.Project.Supervisor, project})
+
+    :ok = Server.Project.Dispatch.register(project, [project_compiled()])
+    assert_receive project_compiled(), 5000
+
+    {:ok, project: project}
+  end
+
+  setup [:with_uri]
+
+  defp with_uri(%{project: project, uri_for: file}) do
+    path = file_path(project, Path.join("lib", file))
+    %{uri: Document.Path.ensure_uri(path)}
+  end
+
+  defp with_uri(_) do
+    :ok
+  end
+
+  def hover_request(path, line, char) do
+    uri = Document.Path.ensure_uri(path)
+
+    params = [
+      text_document: [uri: uri],
+      position: [line: line, character: char]
+    ]
+
+    with {:ok, _} <- Document.Store.open_temporary(uri),
+         {:ok, req} <- build(Requests.Hover, params) do
+      Convert.to_native(req)
+    end
+  end
+
+  def handle(request, project) do
+    Handlers.Hover.handle(request, %Env{project: project})
+  end
+
+  describe "module hover" do
+    @describetag uri_for: "docs.ex"
+
+    test "module with docs", %{project: project, uri: uri} do
+      # alias Project.Docs.PublicModule
+      #                    ^
+      {:ok, request} = hover_request(uri, 64, 21)
```

Comment:
🤔 I would like to think about tests like this a little more; Tests with hard-coded magic numbers (here referring to files in a different place) are extremely brittle and difficult to debug if they fail.

Is there a way we can build a file on the fly and annotate it inside the tests as we do with the `CompletionCase`?

Reply:
Agreed. When I first wrote this test file, I started by copying the go_to_definition_test.exs and then adapting it, because I wasn't yet familiar with what was required to setup the test environment. 

Now that I have a better understanding, I'll take another pass at this.